3scale-operator/tasks/main.yml

```diff
@@ -87,18 +107,63 @@
             adminPortalCredentialsRef:
               name: 3scaleportal
       failed_when: result.resources[0] is not defined
```

Comment:
you forgot to register the result var. you also need to confirm whether the result var will have a resources attribute.